lib/child_process.js

```diff
@@ -161,15 +168,40 @@ function exec(command, options, callback) {
 }
 
 const customPromiseExecFunction = (orig) => {
-  return (...args) => {
+  return (file, options = {}) => {
     let resolve;
     let reject;
     const promise = new Promise((res, rej) => {
       resolve = res;
       reject = rej;
     });
 
-    promise.child = orig(...args, (err, stdout, stderr) => {
+    if (typeof options !== 'object') {
+      return Promise.reject(
+        new ERR_INVALID_ARG_TYPE(
+          'options',
+          'Object',
+          options));
+    }
+
+    const { signal } = options;
+    const hasSignal = signal != null;
+    const isNotValidSignal = typeof signal !== 'object' ||
+      !('aborted' in signal);
+
+    if (hasSignal) {
+      if (isNotValidSignal) return Promise.reject(
+        new ERR_INVALID_ARG_TYPE(
+          'options.signal',
+          'AbortSignal',
+          signal));
+
+      signal.addEventListener('abort', () => {
+        reject(lazyDOMException('AbortError'));
```

Comment:
This should likely also terminate the spawned process in addition to just rejecting the Promise.

Reply:
Calling `promise.child.kill` should be enough or there's something else that I missed?

Reply:
That *should* be but it also raises the question of exactly which signal should be sent. It may be worth establishing a reasonable default but allowing (via an additional option) a different signal to be specified... e.g. 

```
execPromisifed(pwdcommand, { cwd: dir, signal, abortSignal: 'SIGINT' })
```

Reply:
Sure, has sense. Regarding the signal, what about `SIGKILL` or `SIGTERM` as default? `SIGNINT` is coupled to interruption from the keyboard, isn't it?

Reply:
Now that I took a look twice into the implementation, there's already a `killSignal` passed through the `options` on `exec`. Should be that enough or I'm missed?